elasticSnorkel.ipynb

```diff
@@ -0,0 +1,410 @@
+    "https://stackoverflow.com/questions/45537916/elasticsearch-proximity-search-using-2-pre-tokenized-arrays"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of hits \n",
+      "275\n",
+      "Result 1\n",
+      "-------------------\n",
+      "sentence\n",
+      "Lady Bird Johnson was born as Claudia Alta Taylor and married Lyndon B. Johnson in 1934.\n",
+      "Result 2\n",
+      "-------------------\n",
+      "sentence\n",
+      "Alex Gerrard - who's married to footballer Steven Gerrard, 35 - returned from her new US residence to her home town to indulge in a beauty treatment on Thursday.\n",
+      "\n",
+      "Result 3\n",
+      "-------------------\n",
+      "sentence\n",
+      "Just over a year later, Kim Davis and the biological father of her twins, Thomas Dale McIntyre Jr., married in a ceremony on November 11, 2007, at Skye Bridge in Wolfe County.    \n",
+      "Result 4\n",
+      "-------------------\n",
+      "sentence\n",
+      "Williams married Van Veen on Saturday afternoon in a private ceremony at a ranch out West, her rep confirmed to the Los Angeles Times.   \n",
+      "Result 5\n",
+      "-------------------\n",
+      "sentence\n",
+      "Williams married Van Veen on Saturday afternoon in a private ceremony at a ranch out West, her rep confirmed to the Los Angeles Times.\n",
+      "Result 6\n",
+      "-------------------\n",
+      "sentence\n",
+      "Parenthood star Erika Christensen has married cyclist Cole Maness.   \n",
+      "Result 7\n",
+      "-------------------\n",
+      "sentence\n",
+      "Kim Davis then married Joe Davis, a delivery driver, on August 24, 1996, in Morehead by Baptist church minister H.G. Pratt.   \n",
+      "Result 8\n",
+      "-------------------\n",
+      "sentence\n",
+      "Surprise Rebecca Hall married actor Morgan Spector in Saugerties New York on Saturday Sept 26 Us Weeklycan confirm  PHOTOS Costar couples   \n",
+      "Result 9\n",
+      "-------------------\n",
+      "sentence\n",
+      "Girls star Allison Williams married technology entrepreneur Ricky Van Veen in Wyoming on Saturday.   \n",
+      "Result 10\n",
+      "-------------------\n",
+      "sentence\n",
+      "The credits from Monday, Sept. 28's episode of The Big Bang Theory still showed Kaley Cuoco's married name, Kaley SweetingThis article originally appeared on Usmagazine.com: Big Bang Theory Credits\n"
+     ]
+    }
+   ],
+   "source": [
+    "distance = 100\n",
+    "value = \"married\"\n",
+    "candField = \"candidates\"\n",
+    "candTag = \"person\"\n",
+    "sentField=\"sentence\"\n",
+    "result = eSearch.searchBetweenCandidates(candField,candTag,sentField,value,distance)\n",
```

Comment:
Can we make this a bit more general?  What about just calling it something like 'search', and then have some keyword argument (e.g `position`) with values:
* 'entire'
* 'between'
* 'left'
* 'right'

This would then subsume both this and the previous `searchIndex` example!